### PR TITLE
New benchmarks

### PR DESCRIPTION
**Pull request by davidhewitt**
I threw some benchmarks together to evaluate #890. Thought it better to submit them in a separate PR.

### CODE REVIEW

File: benches/bench_pyclass.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    fn __iter__(slf: PyRef<'p, Self>) -> PyResult<Py<Iter>> {
+        let iter = Iter {
+            inner: slf.values.clone().into_iter(),
+        };
+        PyCell::new(slf.py(), iter).map(Into::into)
+    }
+}
+
+#[bench]
+fn create_many_pyclass(b: &mut Bencher) {
```
Comment (kngwyu) [inline benches/bench_pyclass.rs:42]: What this benchmark is for?

File: benches/bench_pyclass.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    b.iter(|| {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let obj = PyCell::new(
+            py,
+            MyClass {
+                obj: py.None(),
+                x: 0,
+                values: (0..10_000).into_iter().collect(),
```
Comment (kngwyu) [inline benches/bench_pyclass.rs:73]: It looks that clippy complains about this.
```suggestion
                values: (0..10_000).collect(),
```

**Comment by davidhewitt** [comment]
👍 for now I've just removed `bench_pyclass`.

I did some benchmarking a while ago in Python using `timeit` where I found that a simple `pyclass` was slower than a simple `dict`. I'd like to add these benchmarks to the pyo3 repo eventually, but really they need to be `.py` benchmarks, so I will add them back another time.

**Comment by kngwyu** [comment]
>  I'd like to add these benchmarks to the pyo3 repo eventually, but really they need to be .py benchmarks, so I will add them back another time.

We have [a benchmark](https://github.com/PyO3/pyo3/tree/master/examples/word-count#benchmark) using pytest. Maybe it's helpful to take a glance at it.

**Comment by kngwyu** [comment]
Thank you!

**Comment by davidhewitt** [comment]
> We have [a benchmark](https://github.com/PyO3/pyo3/tree/master/examples/word-count#benchmark) using pytest. Maybe it's helpful to take a glance at it.

Ah nice, I will be sure to play around with that some time 👍

**Pull request merged**